### PR TITLE
Add stylus programSize and programMemoryFootprint precompiles

### PR DESCRIPTION
**Pull request by joshuacolvin0**
Requires
- https://github.com/OffchainLabs/stylus-contracts/pull/13

### CODE REVIEW

File: system_tests/program_test.go [hunk 1]
```diff
@@ -84,6 +84,16 @@ func keccakTest(t *testing.T, jit bool) {
...
+	programSize, err := arbWasm.ProgramSize(nil, programAddress)
+	Require(t, err)
+	if programSize < 5000 || programSize > 20000 {
+		Fatal(t, "unexpected size", programSize)
+	}
+	programMemoryFootprint, err := arbWasm.ProgramMemoryFootprint(nil, programAddress)
+	Require(t, err)
+	if programMemoryFootprint != 1 {
+		Fatal(t, "unexpected memory footprint", programMemoryFootprint)
+	}
```
Comment (rachel-bousfield) [inline system_tests/program_test.go]: Lets move this to the memory test, which has an example that's known to be 120 pages :)

File: system_tests/program_test.go [hunk 2]
```diff
@@ -84,6 +84,16 @@ func keccakTest(t *testing.T, jit bool) {
 	if otherVersion != programVersion {
 		Fatal(t, "mismatched versions", stylusVersion, programVersion)
 	}
+	programSize, err := arbWasm.ProgramSize(nil, programAddress)
+	Require(t, err)
+	if programSize < 5000 || programSize > 20000 {
+		Fatal(t, "unexpected size", programSize)
+	}
```
Comment (rachel-bousfield) [inline system_tests/program_test.go]: I think we know the exact size and could check against it

File: arbos/programs/programs.go [hunk 3]
```diff
@@ -333,6 +333,17 @@ func (p Programs) CodehashVersion(codeHash common.Hash) (uint16, error) {
...
+	program, err := p.deserializeProgram(codeHash)
+	// wasmSize represents the number of half kb units, return as bytes
+	return uint32(program.wasmSize) * 512, err
+}
+
+func (p Programs) CodehashMemoryFootprint(codeHash common.Hash) (uint16, error) {
+	program, err := p.deserializeProgram(codeHash)
+	return program.footprint, err
+}
+
```
Comment (rachel-bousfield) [inline arbos/programs/programs.go]: Let's name these `ProgramSize` and `ProgramMemoryFootprint`, but still have them take codehashes

**Review by rachel-bousfield — CHANGES_REQUESTED** [review]
Nice work, just some minor things :)

File: system_tests/program_test.go (hunk 1 above)
Comment (joshuacolvin0) [reply system_tests/program_test.go]: :+1:

File: system_tests/program_test.go (hunk 2 above)
Comment (joshuacolvin0) [reply system_tests/program_test.go]: :+1:

File: arbos/programs/programs.go (hunk 3 above)
Comment (joshuacolvin0) [reply arbos/programs/programs.go]: :+1:

**Comment by joshuacolvin0** [comment]
LGTM

**Review by rachel-bousfield — APPROVED** [review]
Looks good!

**Pull request merged**